File: license-tracker-backend/src/main/java/com/prodapt/license_tracker_backend/service/LicenseService.java
package com.prodapt.license_tracker_backend.service;


import com.prodapt.license_tracker_backend.dto.LicenseRequest;
import com.prodapt.license_tracker_backend.dto.LicenseResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface LicenseService {
    LicenseResponse createLicense(LicenseRequest request);
    Page<LicenseResponse> getAllLicenses(Pageable pageable);
    LicenseResponse getLicenseById(Long id);
    LicenseResponse getLicenseByKey(String licenseKey);
    LicenseResponse updateLicense(Long id, LicenseRequest request);
    void deleteLicense(Long id);
}

File: licensetracker_frontend/src/services/licenseService.js
import api from './api';

const licenseService = {
    getAllLicenses: async (page = 0, size = 10, sortBy = 'id') => {
        const response = await api.get(`/licenses?page=${page}&size=${size}&sortBy=${sortBy}`);
        return response.data;
    },

    getLicenseById: async (id) => {
        const response = await api.get(`/licenses/${id}`);
        return response.data;
    },

    getLicenseByKey: async (key) => {
        const response = await api.get(`/licenses/by-key/${key}`);
        return response.data;
    },

    createLicense: async (licenseData) => {
        const response = await api.post('/licenses', licenseData);
        return response.data;
    },

    updateLicense: async (id, licenseData) => {
        const response = await api.put(`/licenses/${id}`, licenseData);
        return response.data;
    },

    deleteLicense: async (id) => {
        const response = await api.delete(`/licenses/${id}`);
        return response.data;
    }
};

export default licenseService;
